src/nvim/os/env.c

```diff
@@ -410,14 +410,14 @@ static char *remove_tail(char *p, char *pend, char *name)
 /// @param[out] mustfree Ouput parameter for the caller to determine if they are
 ///           responsible for releasing memory. Must be initialized to false
 ///           by the caller.
-char_u *vim_getenv(char_u *name, bool *mustfree)
+char *vim_getenv(char *name, bool *mustfree)
 {
-  char_u      *p;
-  char_u      *pend;
+  char *p;
+  char *pend;
   int vimruntime;
 
 
-  p = (char_u *)os_getenv((char *)name);
+  p = (char *)os_getenv(name);
```

Comment:
Okay.  I'll get working on merging my other branch with this one, and push it in later.  

I would definitely agree that a patch shouldn't introduce incorrect behavior or break things that were right, as a fundamental principle.  For my own understanding --  can you help me see how this does that?  I went back and forth on this, but ended up going this way because the cast to (char_u*) that was there *seemed* to me to do the same thing, so it appeared to me like my changes were a lateral move, rather than a step back.